wdl_runner/file_util.py

```diff
@@ -9,96 +9,142 @@
 # file_util.py
 
 import logging
-import simplejson
 import string
-import subprocess
+from pathlib import Path
+from typing import List
+from urllib.parse import urlparse
 
+import simplejson
+from google.cloud import storage
 from googleapiclient import discovery
 from googleapiclient.errors import HttpError
 
-import sys_util
+import wdl_runner.sys_util
 
 
 def file_safe_substitute(file_name, mapping):
-  """Performs placeholder replacement on a file, saving contents in place."""
-
-  with open(file_name, 'r') as f:
-    file_contents = f.read()
-    return string.Template(file_contents).safe_substitute(mapping)
-
-
-def gsutil_cp(source_files, dest_dir):
-  """Copies files to GCS and exits on error."""
-
-  cp_cmd = ['gsutil', 'cp'] + source_files + [dest_dir]
-
-  logging.info("Copying %s to %s", source_files, dest_dir)
-
-  # Copies can fail, so include retries...
-  for attempt in range(3):
-    p = subprocess.Popen(cp_cmd, stderr=subprocess.PIPE)
-    return_code = p.wait()
-    if not return_code:
-      return
-
-    logging.warn("Copy %s to %s failed: attempt %d",
-                 source_files, dest_dir, attempt)
-
-  sys_util.exit_with_error(
-      "copying files from %s to %s failed: %s" % (
-          source_files, dest_dir, p.stderr.read()))
+    """Performs placeholder replacement on a file, saving contents in place."""
+
+    with open(file_name, "r") as f:
+        file_contents = f.read()
+        return string.Template(file_contents).safe_substitute(mapping)
+
+
+def _upload_one_file(client, src_file, dest_bucket, dest_gs_url):
```

Comment:
Very nitpicky but would be good to have some type annotation for these arguments for future developers

Reply:
Added ; that's not nitpicky.